File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.java
class Solution {
    public int[][] divideArray(int[] nums, int k) {
        Arrays.sort(nums);
        int n=nums.length;
        int n3 = n/3;
        int[][] ans = new int[n3][3];
        for(int i=0;i<n3;i++) {
            if(nums[3*i+2]-nums[3*i]>k)
                return new int[0][0];
            ans[i]=new int[]{nums[3*i],nums[3*i+1],nums[3*i+2]};
        }
        return ans;
    }
}

File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.cpp
class Solution {
public:
    vector<vector<int>> divideArray(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        int n3 = n/3;
        for(int i=0;i<n3;i++) {
            if(nums[3*i+2]-nums[3*i]>k)
                return vector<vector<int>>();
        }
        vector<vector<int>> ans;
        for(int i=0;i<n3;i++) {
            ans.push_back(vector<int>{nums[3*i],nums[3*i+1],nums[3*i+2]});
        }
        return ans;
    }
};